pkg/executor/openstack.go

```diff
@@ -0,0 +1,561 @@
+// Execute creates a instance and runs the provided command in it. When the command completes, the instance
+// is stopped i.e. the container is not restarted automatically.
+func (os Openstack) Execute(command string) (TaskHandle, error) {
+	provider, err := openstack.AuthenticatedClient(os.config.Auth)
+	if err != nil {
+		return nil, errors.Wrapf(err, "couldn't get provider")
+	}
+
+	os.client, err = openstack.NewComputeV2(provider, gophercloud.EndpointOpts{Region: "RegionOne"})
+	if err != nil {
+		return nil, errors.Wrap(err, "couldn't get compute client")
+	}
+
+	flavorID, err := os.ensureFlavor()
+	if err != nil {
+		return nil, errors.Wrap(err, "couldn't ensure flavor")
+	}
+
+	if err = os.ensureKeypair(); err != nil {
+		return nil, errors.Wrap(err, "couldn't ensure keypair")
+	}
+
+	floatingIP, err := os.findFloatingIP()
+	if err != nil {
+		return nil, errors.Wrap(err, "couln't find floating IP")
+	}
+
+	image, err := images.IDFromName(os.client, os.config.Image)
+
+	instanceName := fmt.Sprintf("krico.%s", uuid.New())
+
+	serverOpts := servers.CreateOpts{
+		Name:      instanceName,
+		FlavorRef: flavorID,
+		ImageRef:  image,
+	}
+
+	serverOptsExt := keypairs.CreateOptsExt{
+		CreateOptsBuilder: serverOpts,
+		KeyName:           kricoKeypairName,
+	}
+
+	instance, err := servers.Create(os.client, serverOptsExt).Extract()
+
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to create instance")
+	}
+
+	log.Debugf("Scheduled instance %s creation", instance.ID)
+
+	err = servers.WaitForStatus(os.client, instance.ID, "ACTIVE", 60)
+
+	if err != nil {
+		return nil, errors.Wrap(err, "couldn't launch instance")
+	}
+
+	log.Debugf("Launched instance %s", instance.ID)
+
+	associateOpts := floatingips.AssociateOpts{
+		FloatingIP: floatingIP,
+	}
+
+	err = floatingips.AssociateInstance(os.client, instance.ID, associateOpts).ExtractErr()
+	if err != nil {
+		return nil, errors.Wrap(err, "couldn't associate floating ip to instance")
+	}
+
+	log.Debugf("Associated %q floating ip", floatingIP)
+
+	remoteConfig := RemoteConfig{
+		User:    os.config.User,
+		KeyPath: os.config.SSHKeyPath,
+		Port:    22,
+	}
+
+	log.Debug("Waiting for instance to boot up")
+
+	// Wait while to ensure that everything booted up!
+	time.Sleep(time.Second*bootUpTimeOut)
+
+	type serverAttributesExt struct {
+		servers.Server
+		extendedserverattributes.ServerAttributesExt
+	}
+
+	var extendedAttributes serverAttributesExt
+
+	err = servers.Get(os.client, instance.ID).ExtractInto(&extendedAttributes)
+	if err != nil {
+		return nil, errors.Wrap(err, "couldn't get instance extended attributes")
+	}
+
+	os.config.HypervisorInstanceName = extendedAttributes.InstanceName
+
+	remote, err := NewRemote(floatingIP, remoteConfig)
+	if err != nil {
+		return nil, errors.Wrap(err, "couldn't create remote executor")
+	}
+	log.Debug("Created remote executor")
+
+	_, err = remote.Execute(command)
+	if err != nil {
+		return nil, errors.Wrap(err, "couldn't start remote task")
```

Comment:
Log the error, but do not wrap it. We do not need to wrap out own errors: they are wrapped already or created using [errors.New()](https://godoc.org/github.com/pkg/errors#example-New). Add information about instance ID and its IP address too, please.